src/mantine-dates/src/components/TimeInputBase/AmPmInput/AmPmInput.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useRef, forwardRef } from 'react';
+import { useMergedRef } from '@mantine/hooks';
+import { MantineSize } from '@mantine/core';
+import useStyles from './AmPmInput.styles';
+
+interface AmPmSelectProps
+  extends Omit<React.ComponentPropsWithoutRef<'input'>, 'onChange' | 'size'> {
+  /** Called with onChange event */
+  onChange(value: string, triggerShift: boolean): void;
+
+  /** Called when input loses focus, used to format value */
+  setValue(value: string): void;
+
+  /** Colon text size */
+  size?: MantineSize;
+
+  /** Displays error message after input */
+  error?: React.ReactNode;
+}
+
+export const AmPmInput = forwardRef<HTMLInputElement, AmPmSelectProps>(
+  (
+    {
+      className,
+      onChange,
+      onFocus,
+      setValue,
+      size = 'sm',
+      error,
+      value,
+      ...others
+    }: AmPmSelectProps,
+    ref
+  ) => {
+    const { classes, cx } = useStyles({ size });
+    const inputRef = useRef<HTMLInputElement>();
+
+    const handleFocus = (event: React.FocusEvent<HTMLInputElement>) => {
+      typeof onFocus === 'function' && onFocus(event);
+      inputRef.current.select();
+    };
+
+    const handleClick = (event: React.MouseEvent<HTMLInputElement, MouseEvent>) => {
+      event.stopPropagation();
+      inputRef.current.select();
+    };
+
+    return (
+      <input
```

Comment:
AmPm input issues:
- When user presses up/down arrows, value should change. For example, if user presses up or down arrow and am is selected it should change to pm, same for the case when pm is selected
- There is no need to handle free input, when user types `a` field should automatically autocomplete to `am`, same for the `p`